### ISSUE TITLE
find/findAll produces unexpected results with stubbed components

### ISSUE THREAD

**Issue by felixlublasser** [vuejs/vue-test-utils#221]
Consider this component:

```xml
<template>
  <div id='ComponentWithChild'>
    <div class='first-child'>
      <child></child>
    </div>
    <div class='second-child'>
      <child></child>
    </div>
  </div>
</template>
```

and this test:

```javascript
import { shallow } from 'vue-test-utils'
import ComponentWithChild from '@/components/ComponentWithChild'

const ChildStub = {
  name: 'child-stub',
  render: () => {}
}

const wrapper = shallow(ComponentWithChild, { stubs: { 'child': ChildStub } })

it ('', () => {
  expect(wrapper.findAll(ChildStub).length).to.equal(2) // true
  expect(wrapper.find('.first-child').findAll(ChildStub).length).to.equal(1) // nope, it's 2
})
```

I would have expected the second expectation to pass, but instead, `findAll` returns both instances of the `ChildStub` component, not just the one in the `.first-child` div.

**Comment by eddyerburgh** [comment]
Yes, this should be fixed.

**Comment by eddyerburgh** [comment]
I've added a fix to the dev branch. It will be released in 1.0.0-beta.7

**Comment by felixlublasser** [comment]
Awesome, thanks a bunch for the quick fix!
